docs/COMPONENTS.rst

```diff
@@ -0,0 +1,67 @@
+===================
+Ceph Lab Components
+===================
+
+The distinct hardware/software components that a lab is composed of and the way
+they interact between them is described here. In general, a lab is composed of
+a scheduler (a.k.a. master node), worker(s), package builder
+(`gitbuilder <http://ceph.com/gitbuilder.cgi>`__), job database (`paddles
+<https://github.com/ceph/paddles>`__), job archive, a web UI (`pulpito
+<https://github.com/ceph/pulpito>`__) and test nodes.
+
+|
+|
+
+.. image:: cephlab.png
+    :align: center
+    :alt: Components of a Ceph Lab. Original at https://docs.google.com/presentation/d/1qA4137X-VB9gnFhZCmMLYudfMJ2_BUJFczviHVbWnpo
+
+|
+|
+
+In the figure above, every service appears on a separate machine but this is
+not a requirement (see :ref:`lab_setup` for an alternative setup). Jobs are
+submitted to the scheduler, which are then processed by workers. One worker
+processes and keeps track of a job (one at a time). The output of the job (logs
+and files associated to an execution) is stored in the archive, which
+is a path in the file system where the worker is running. The job database
+contains information about the status of jobs and test nodes, as well as
+results of executions (# of tests passed, failed, etc.). All this information
+can be visualized in ``pulpito``, the web UI. For an example, see Red Hat's Lab
```

Comment:
While it's owned and operated by Red Hat, that is specifically the "Ceph community lab"